### ISSUE TITLE
geospatial data

### ISSUE THREAD

**Issue by gurdeep101** [antklen/diameter-clustering#4]
is it possible to use this for geospatial data by precomputing a distance matrix for lat-long values based on haversine distance?

**Comment by antklen** [comment]
Yes, definitely it is possible. If you have precomputed distance matrix, you can make clustering with this code:
```python
DISTANCE_THRESHOLD = 1  # set your distance threshold
model = MaxDiameterClustering(max_distance=DISTANCE_THRESHOLD, sparse_dist=False, precomputed_dist=True)
labels = model.fit_predict(dist_matrix)
```
Don't forget to set `sparse_dist=False` if your distance matrix is ordinary numpy array (not sparse matrix).

I haven't tried this clustering with geospatial data, but it looks like very suitable use case. Having maximum distance between points in cluster makes a lot of sense for geo data.

Feel free to ask if you have any questions.

**Comment by gurdeep101** [comment]
thanks. I'll try this out with the above suggestions and compare with the R library. Will let you know.

**Comment by gurdeep101** [comment]
ok so by and large we're getting results for LeaderClustering with the library that are similar to the R implementation. I did this by defining a custom function "my_hav" to calculate the haversine distance and passing it to the model parameter - see sample code below. Turns out didn't need to precompute a distance matrix.

The only request would be if it is possible to add functionality to return cluster centroids for each cluster as well. This helps especially if one needs to plot the points and draw a circle around them

`
import numpy as np
import diameter_clustering as dc
import haversine as hs

coords = [lat, long]. to_numpy()

def my_hav(c1, c2):
    return hs.haversine(c1, c2)

max_rad = 1
model = dc.LeaderClustering(max_radius = max_rad, metric = my_hav, deterministic = True)
labels = model.fit_predict(coords)
`

**Comment by antklen** [comment]
It is interesting that you don't need to precompute distance matrix, I missed this opportunity :) 

May be I will be able to add cluster centroids for each cluster next week.
As for now, after model was fitted it has `leaders_` attribute. It contains array with 1 for points which are cluster centroids and with 0 for all other points. Unfortunately there is no information which cluster each centroid belongs to. But I think it can help with drawing centroids and circles around them.

**Comment by supriyabachal** [comment]
hi! @gurdeep101  how many points did you use while clustering and max_rad=1 is in miles or kms or radians?
